### ISSUE TITLE
Helldivers 2 not working

### ISSUE THREAD

**Issue by Faradayff** [adirh3/Sonar.AutoSwitch#28]
With Helldivers 2 doesn't change the profile, the window title have a special character, is "HELLDIVERS™ 2" but I have tried using the executable name "helldivers2" and it doesn't change either.

**Comment by Faradayff** [comment]
I've been researching, looking into de code and debugging and I have realized the code doesn't detect when you change to the window of helldivers2, but I can't understand why..
I have tried to change the CharSet.Auto to CharSet.Unicode but doesn't change anything, seems to be something with the game itself, not the special characters on the name

**Comment by jlopez994** [comment]
Same here. I've trying too but there is no way to get the profile updated with this game. Is there still support for this project?

**Comment by Mativec** [comment]
I think it would be useful to match the title with regexp so we can use * or others to avoid special character 🤔 

I will look at it when I got time

**Comment by Faradayff** [comment]
Thank you for taking attention to this. The problem is not character related, debugging, when I change every window, the code detects it, but not with helldivers 2, the code is not aware that I changed to helldivers window

**Comment by Mativec** [comment]
Indeed, it's disturbing 🤔 
Even with a regex as the window title (my next PR) it doesn't switch

**Comment by Mativec** [comment]
@adirh3 I thinks the app have trouble catching the pid, if you have an idea (I will look at it this week)

My debug on the code (Win32WindowEventManager.cs) :
```csharp
private void WindowEventCallback(nint hWinEventHook, uint eventType, nint hwnd, int idObject,
    int idChild, uint dwEventThread, uint dwmsEventTime)
{
    string windowTitle = GetWindowTitle(hwnd);
    Debug.WriteLine("DEBUG : " + windowTitle);
    if (GetWindowThreadProcessId(hwnd, out var pid) == 0)
        return;
    Debug.WriteLine("COUCOU : ");
    string? exeName = null;
```
Log :
```sh
DEBUG : 
COUCOU : 
DEBUG : 
COUCOU : 
DEBUG : HELLDIVERS™ 2
DEBUG : Sonar.AutoSwitch (Running) - Microsoft Visual Studio
COUCOU : 
```

Ps : "COUCOU" means "HELLO" in french and is my debug word 😂

**Comment by Faradayff** [comment]
You got it right in the point, this sounds promising

**Comment by Mativec** [comment]
I got it @Faradayff 

![Image](https://github.com/user-attachments/assets/05d33ee5-dc08-451b-a97e-05aa1cc33028)

Let me a moment to test if I broke something and I will make a PR 😄

edit : I did broke something 😂

**Comment by Faradayff** [comment]
Yes!!! You did it!! Where was the key? the pid? How did you managed to get it?
I hope you can fix the new error soon.
It's been a year using helldivers as default profile cause it won't switch hahahah

**Comment by Mativec** [comment]
To get the window, I used ``helldivers*`` to avoid the ``TM`` special character.
Then, the software couldn't get the name of the executable with the PID but it still got the right windowTitle.
So instead of stopping, I tell it to continue with only the windowTitle.
everything seems to work now, I'm creating the PR.

edit : typo

**Comment by Faradayff** [comment]
Ouh, that's interesting
Good job this is really great!! 
Thank you mate!!
